### PR TITLE
editoast: forward core error type

### LINKED ISSUE #5244: editoast: core errors are forwarded with very little context
### What happened?

Core errors don't include the context when forwarded from editoast. 

For example, core may send a response such as: 

```json
{
    "context": {
        "stack_trace": [
            "STDCMStandardAllowance.java:60",
            "STDCMPostProcessing.java:48",
            "STDCMPathfinding.java:79",
            "STDCMEndpoint.java:95",
            "FkRegex.java:153",
            "FkRegex.java:217",
            "FkChain.java:72",
            "TkFork.java:98",
            "TkFallback.java:84",
            "TkFallback.java:66",
            "TkWrap.java:58",
            "TkSlf4j.java:110",
            "BkBasic.java:123",
            "BkBasic.java:99",
            "BkSafe.java:46",
            "BkWrap.java:51",
            "BkParallel.java:81",
            "ThreadPoolExecutor.java:1136",
            "ThreadPoolExecutor.java:635",
            "Thread.java:833"
        ],
        "message": "conflict offset is already on a range transition"
    },
    "message": "assert check failed",
    "type": "assert_error"
}
```

editoast will send a response looking like this: 

```json
{
    "type": "editoast:coreclient:Forward",
    "context": {
        "status": 500,
        "core_error": {
            "cause": null,
            "message": "assert check failed",
            "trace": null,
            "type": "assert_error"
        },
        "url": "http://localhost:8080/stdcm"
    },
    "message": "assert check failed"
}
```

The actual message of the assertion error is lost.

### What did you expect to happen?

More context to be included, at least the actual assert error

### How can we reproduce it (as minimally and precisely as possible)?

The simplest way to reproduce would be to edit core to add an `assert false : "foobar";` line somewhere, and look for `foorbar` in the response from editoast

### What operating system, browser and environment are you using?

- **Browser**: Firefox
- **OS**: ubuntu
- **Env**: Local

### OSRD version (top right corner `Account` button > `Informations`)

12f29b0f150197c6eff3aa9679095b1f1b83ba55

### LINKED ISSUE #5350: editoast: core error forwarding doesn't include its type anymore (empty string)
### What happened?

Core had an error that was forwarded this way: `{"type":"","context":{},"message":"No path could be found"}`. `type` should have been `no_path_found` instead of an empty string. 
This change breaks [this check](https://github.com/osrd-project/osrd/blob/dev/tests/fuzzer/fuzzer.py#L184) in the fuzzer, which makes the integration tests unstable.

### What did you expect to happen?

_No response_

### How can we reproduce it (as minimally and precisely as possible)?

Trigger an error in core, easiest one is an impossible path

### What operating system, browser and environment are you using?

- **Browser**: Firefox
- **OS**: windows
- **Env**: Local

### OSRD version (top right corner `Account` button > `Informations`)

7c02b7b996f82c6db461fb207d460ebd239eae0f

### PR DESCRIPTION
**Pull request by Khoyo**
Fixes #5350.
Fixes #5244.

### CODE REVIEW

File: editoast/src/core/mod.rs [hunk 1]
```diff
@@ -291,13 +282,6 @@ enum CoreError {
     #[error("Cannot parse Core response: {msg}")]
     #[editoast_error(status = 500)]
     CoreResponseFormatError { msg: String },
-    /// A standard core error was found in the response, so it is forwarded
```
Comment (leovalais) [inline editoast/src/core/mod.rs:294]: Could you add back that variant with `core_error: StandardCoreError`?

File: editoast/src/core/mod.rs [hunk 2]
```diff
@@ -307,6 +291,8 @@ enum CoreError {
         url: String,
         raw_error: String,
     },
+    #[error("Cannot convert core error to a UTF-8 string")]
```
Comment (leovalais) [inline editoast/src/core/mod.rs]: You removed the forwarding with a raw utf8 string so that message is wrong

**Review by leovalais — CHANGES_REQUESTED** [review]
LGTM & tested. @Tguisnet could you review this as well since I worked a little on this?

File: editoast/src/core/mod.rs (hunk 1 above)
Comment (Khoyo) [reply editoast/src/core/mod.rs:294]: I don't get what you mean

Comment (leovalais) [reply editoast/src/core/mod.rs:294]: Let's keep it this :sadge:

**Review by leovalais — APPROVED** [review]
LGTM, @Tguisnet is reviewing it

**Review by Tguisnet — APPROVED** [review]
LGTM

**Pull request merged**